hibernate-core/src/main/java/org/hibernate/loader/BatchLoadSizingStrategy.java

```diff
@@ -0,0 +1,16 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.loader;
+
+/**
+ * Strategy (pluggable) for determining an optimal size for batch loads.
+ *
+ * @author Steve Ebersole
+ */
+public interface BatchLoadSizingStrategy {
+	int determineOptimalBatchLoadSize(int numberOfKeyColumns, int numberOfKeys);
```

Comment:
Its not part of the JavaDoc because as I stated in the PR description this is intended just for us to discuss.  Do you typically fill out JavaDoc for such prototyping?

Well `numberOfKeyColumns` is important to be able to understand how many parameters this translates to in the SQL call for databases that limit the numer of parameters.  Loading 1000 entities whose id is made up of 1 column requires 1000 parameters, but loading 1000 entities whose id is made up of 2 column requires 2000 parameters... Big difference.

The idea of this is for the Dialect to be able to influence the batch size when we were not told a size by the user as discussed in the Jira.

Look, this is a prototype meant for us to discuss the design.  So if you have something you'd like to see different that is the point of this.  But really *constructive* criticism is usually better.  So maybe rather than just pointing out what you don't like you could propose an alternative?  But in terms of this contract you are just missing the point of it (as it was discussed in the Jira).

As for the user being able to tell us the batch size, the discussion for that (again, on the Jira) was to use `@BatchSize` (which is entity specific).  Again, if you don't like that then propose a way for that to happen.  However I will say that overloading these methods to accept a "batchSize" is not something I would vote for; thats classic implementation details leaking through an API.